### PR TITLE
fix(data-service): use databases where user has built in roles in `listDatabases` COMPASS-5691

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-5691 https://github.com/mongodb-js/compass/issues/2953

## Description
Atlas shared tier clusters (serverless, M0, M2) with MongoDB server version 5+ do not currently return privileges with the `connectionStatus` command response (fix might be in soon/already w/ [CLOUDP-115805](https://jira.mongodb.org/browse/CLOUDP-115805)). It also fails to `listDatabases`. As a result, currently a user with permissions to only one database would not be able to see their database. 

While there may be a release of Atlas soon making this issue go away, the changes in this PR introduce a workaround so users with exclusive database permissions can use their databases. These changes expand our extra `listDatabases` affordances by referencing the [authenticatedUserRoles](https://www.mongodb.com/docs/manual/reference/command/connectionStatus/#mongodb-data-connectionStatus.authinfo.authenticatedUserRoles) in the `connectionStatus` response. We then add all of the databases where a user has any of the `read`, `readWrite`, `dbAdmin`, `dbOwner` [built-in roles](https://www.mongodb.com/docs/manual/reference/built-in-roles/).

## Motivation and Context
- [x] Bugfix

## Open Questions
Is there another way to get this information? Is there anything we can do for cases where users are using custom roles with custom privileges for their access management?

## Types of changes
- [x] Patch (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/data-service/src/data-service.ts [hunk 1]
```diff
@@ -500,16 +520,33 @@ class DataService extends EventEmitter {
...
+          listDatabases(),
+          getDatabasesFromPrivileges(),
+          getDatabasesFromRoles(),
+        ]);
 
       const databases = uniqueBy(
         // NB: Order is important, we want listed collections to take precedence
         // if they were fetched successfully
-        [...databasesFromPrivileges, ...listedDatabases],
+        [...databasesFromRoles, ...databasesFromPrivileges, ...listedDatabases],
```
Comment (lerouxb) [inline packages/data-service/src/data-service.ts:549]: I'm trying to make sense of the comment that was here before. Where's listed collections coming from?

**Review by lerouxb — APPROVED** [review]
It will be interesting to see if we can get a release out with this fix before cloud just fixes it.

**Comment by lerouxb** [comment]
The ubuntu failure is a flake (something about the timing of that leafygreen component I don't understand well enough to reliably work around) and the others are failing due to 5.3. Should be fixed by https://github.com/mongodb-js/compass/pull/2937

**Comment by lerouxb** [comment]
I updated the branch so long to pull in the e2e fix for 5.3

File: packages/data-service/src/data-service.ts (hunk 1 above)
Comment (mcasimir) [reply packages/data-service/src/data-service.ts:549]: I'm assuming this is a merge of databases taken from the privileges and the result of `adminDb.command({listDatabases: ...})` + `adminDb.command({listCollections: dbName})`, probably the db objects returned by the commands are more accurate than the privileges one

**Pull request merged**